packages/invoices/src/OutboundShipment/DetailView/modals/ItemDetailsModal.tsx

```diff
@@ -206,32 +212,20 @@ export const ItemDetailsModal: React.FC<ItemDetailsModalProps> = ({
     onReset();
   };
   const upsert = () => {
-    if (!summaryItem) return;
-
-    const values = getValues();
-    const invoiceLines = batchRows.map(batch =>
-      getInvoiceLine(
-        generateUUID(),
-        summaryItem,
-        batch,
-        Number(values[batch.id] || 0)
-      )
-    );
+    if (!summaryItem) return null;
 
-    invoiceLines
-      .filter(line => line.numberOfPacks > 0)
-      .forEach(upsertInvoiceLine);
-    const placeholderValue = Number(values['placeholder'] || 0);
-    if (placeholderValue > 0) {
-      invoiceLines.push(
-        getInvoiceLine(
-          'placeholder',
-          summaryItem,
-          { id: 'placeholder', expiryDate: '' },
-          placeholderValue
-        )
+    // TODO: Handle placeholder upserting.
+    const invoiceLines = batchRows
+      .filter(({ id }) => id !== 'placeholder')
+      .map(batch =>
+        getInvoiceLine(generateUUID(), summaryItem, batch, batch.quantity)
```

Comment:
I've just handled the zero quantity invoice lines in the reducer, as they might have been edited to zero quantity rather than not being touched. So, rather than track lines which have been touched on top of the quantity, just deriving if they already exist and deleting them